errors_nounix.go

```diff
@@ -0,0 +1,15 @@
+//go:build !unix
+// +build !unix
+
+package fsutil
```

Comment:
Wondering if these should live in an internal package, as (I think) we don't want consumers to be using these to match against (they're only do prevent having to split out the implementation further, correct?)

Reply:
What's the point of returning typed errors if we don't want consumers to match against them?

I don't really see an issue with old code as well, but if we define these separately, then we should export them as well to let consumers no what they can match against.